scripts/pythondistdeps.py

```diff
@@ -18,6 +18,96 @@
 from warnings import warn
 
 
+class RpmVersion():
+    def __init__(self, version_id):
+        version = parse_version(version_id)
+        if isinstance(version._version, str):
+            self.version = version._version
+        else:
+            self.epoch = version._version.epoch
+            self.version = list(version._version.release)
+            self.pre = version._version.pre
+            self.dev = version._version.dev
+            self.post = version._version.post
+
+    def increment(self):
+        self.version[-1] += 1
+        self.pre = None
+        self.dev = None
+        self.post = None
+        return self
+
+    def __str__(self):
+        if isinstance(self.version, str):
+            return self.version
+        if self.epoch:
+            rpm_epoch = str(self.epoch) + ':'
+        else:
+            rpm_epoch = ''
+        while self.version[-1] == 0:
+            self.version.pop()
+        rpm_version = '.'.join(str(x) for x in self.version)
+        if self.pre:
+            rpm_suffix = '~%s' % ''.join(str(x) for x in self.pre)
+        elif self.post:
+            rpm_suffix = '^post%d' % self.post[1]
+        else:
+            rpm_suffix = ''
+        return '%s%s%s' % (rpm_epoch, rpm_version, rpm_suffix)
+
+def convert_compatible(name, operator, version_id):
+    if version_id.endswith('.*'):
+        return 'Invalid version'
```

Comment:
What does RPM say in this case? Does it fail the build saying version cannot have spaces? Should this script exit with 1 instead, with a meaningful error message?

Reply:
The expression is invalid, and will cause setup.py to fail, so I'm reasonably certain that we'd never see it, but exiting with an error code is probably a good idea anyway.

(rpm would otherwise require a package named "Invalid version")

Reply:
I've pushed a new commit.  The script now exits in these cases.